### DISCUSSION TITLE
sed: can't read /etc/update-manager/release-upgrades: No such file or directory

### DISCUSSION THREAD

**Discussion by haroldjeshua** [microsoft/WSL#9495] [General]
### Version

Microsoft Windows [Version 10.0.19044.2486]

### WSL Version

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

Ubuntu-22.04

### Distro Version

Ubuntu-22.04

### Repro Steps

sed: can't read /etc/update-manager/release-upgrades: No such file or directory

### Expected Behavior

After opening the terminal or running it, it should be directed to the root directory.

### Actual Behavior

![image](https://user-images.githubusercontent.com/34917956/212799893-a92c3f73-a1ec-4ee9-952f-eb67faf23558.png)

At startup, sometimes it results to this. Then after a while, like an hour or so, WSL comes back normal and operational.

### Diagnostic Logs

I installed first Ubuntu 22.04 on the Microsoft Store early mid last year and it was stable until Dec, it starts to reproduce the error above. Then I run  `wsl --list --online` and still doesn't show 22.04
![image](https://user-images.githubusercontent.com/34917956/212800629-1986cab0-c0e3-4369-a62e-edb733a8dc8d.png)

**Comment by elsaco** [comment]
@haroldjeshua `Ubuntu` in `wsl --list --online` output is `Ubuntu 22.04`. It is defined in `DistributionInfo.json`. See details at https://github.com/microsoft/WSL/blob/master/distributions/DistributionInfo.json#L4

**Comment by grayhound** [comment]
@haroldjeshua Do you have Docker installed?

Try to run 

`wsl --list`

It will show everything installed on your system.

If you have docker you may see something like this:

```
docker-desktop-data (Default)
docker-desktop
Ubuntu-22.04
```

In my case `docker-desktop-data` was by default which is wrong. I couldn't run docker and had the same error as yours trying to log in into Ubuntu. Exactly the same error.

To fix this run:

`wsl --set-default Ubuntu-22.04`

and then try to log in to your Ubuntu.

This is just a suggestion, your case maybe a little bit different.

The main idea - check that your Ubuntu-22.04 is a default one.

**Comment by MartyMcFlyInTheSky** [comment]
I got the same error after tinkering with the wsl.conf file under /etc (don't know if its related though...). As I saw it, my user is gone and there's just a default user. I can't even sudo su because 'sudo' is not recognised as a command. Something is utterly broken. I'm reinstalling the distro now, as idk how to recover from that.

**Comment by januaraliosada** [comment]
I fix it by restarting my pc

**Comment by quique0194** [comment]
I fixed it by restarting wsl only:

```
wsl --shutdown
```

**Comment by prithvirajaramesh** [comment]
Restart Your Pc it will Work
